daisy_workflows/image_import/import_image.sh

```diff
@@ -48,24 +54,50 @@ function resizeDisk() {
   fi
 }
 
-# Mount GCS bucket containing the disk image.
-mkdir -p /gcs/${SOURCEBUCKET}
-gcsfuse --implicit-dirs ${SOURCEBUCKET} /gcs/${SOURCEBUCKET}
+function copyImageToScratchDisk() {
+  local sourceSizeGigabytes="$((1 + (SOURCE_SIZE_BYTES - 1) / BYTES_1GB))"
+  local scratchDiskSizeGigabytes=sourceSizeGigabytes
+  # We allocate double capacity for OVA, which would
+  # require making an additional copy of its enclosed VMDK.
+  if [[ "${SOURCE_FILE_TYPE}" == "ova" ]]; then
+     scratchDiskSizeGigabytes=$((sourceSizeGigabytes * 2))
+  fi
+
+  # This disk is initially created with 10GB of space.
+  # Enlarge it if that's insufficient to hold the input image.
+  if [[ ${scratchDiskSizeGigabytes} -gt 10 ]]; then
+    resizeDisk "${SCRATCH_DISK_NAME}" "${scratchDiskSizeGigabytes}" "${ZONE}"
+  fi
+
+  local scratchPath="/daisy-scratch/$(dirname ${SOURCEPATH})/"
+  mkdir -p "${scratchPath}"
+  # /dev/sdb is used since the scratch disk is the second
+  # disk that's attached in import_disk.wf.json.
+  mkfs.ext4 /dev/sdb
+  mount /dev/sdb "${scratchPath}"
+  if [[ $? -ne 0 ]]; then
+    echo "ImportFailed: Failed to prepare scratch disk."
+  fi
+  gsutil cp "${SOURCEURL}" "${scratchPath}"
+  echo "Import: Copied image from ${SOURCEURL} to ${scratchPath}"
+}
+
+copyImageToScratchDisk
 
-# Atrocious OVA hack.
-SOURCEFILE_TYPE="${$SOURCE_DISK_FILE##*.}"
-if [[ "${SOURCEFILE_TYPE}" == "ova" ]]; then
+# If the image is an OVA, then copy out its VMDK.
+if [[ "${SOURCE_FILE_TYPE}" == "ova" ]]; then
   echo "Import: Unpacking VMDK files from ova."
-  VMDK="$(tar --list -f /gcs/${SOURCEPATH} | grep -m1 vmdk)"
-  tar -C /gcs/${DAISY_SOURCE_URL#"gs://"} -xf /gcs/${SOURCEPATH} ${VMDK}
+  VMDK="$(tar --list -f /daisy-scratch/${SOURCEPATH} | grep -m1 vmdk)"
+  tar -C /daisy-scratch/${DAISY_SOURCE_URL#"gs://"} -xf /daisy-scratch/${SOURCEPATH} ${VMDK}
   SOURCEPATH="${DAISY_SOURCE_URL#"gs://"}/${VMDK}"
   echo "Import: New source file is ${VMDK}"
 fi
 
+# Ensure the output disk has sufficient space to accept the disk image.
 # Disk image size info.
-SIZE_BYTES=$(qemu-img info --output "json" /gcs/${SOURCEPATH} | grep -m1 "virtual-size" | grep -o '[0-9]\+')
+SIZE_BYTES=$(qemu-img info --output "json" /daisy-scratch/${SOURCEPATH} | grep -m1 "virtual-size" | grep -o '[0-9]\+')
  # Round up to the next GB.
-SIZE_GB=$(awk "BEGIN {print int((${SIZE_BYTES}/1000000000)+ 1)}")
+SIZE_GB="$((1 + (SOURCE_SIZE_BYTES - 1) / BYTES_1GB))"
```

Comment:
why this change? arithmetic expansion only supports integers, so division cases are risky.

Reply:
> why this change?

@dntczdx's asked for it earlier (one billion bytes isn't a gigabyte, and by not subtracting one we will allocate an extra gigabyte to the image if the input image was a multiple of one gigabyte already)

> arithmetic expansion only supports integers, so division cases are risky.

Interesting, risky in what sense? (The previous usage of `int` performs the same rounding that is used here)

Reply:
yes, but the cast to integer happens after the division operation, so it's safe bc it's being applied intentionally and as a step. with bash/posix arithmetic, it simply doesn't support floats, so you get undesired behavior easily. it's maybe just an aversion to bash arithmetic developed over my time as a bash dev

Reply:
@hopkiw I am a bit confused. This calculation doesn't rely on float. Oppositely, it intentionally rely on how int is divided. In which case it works unexpected?